### PR TITLE
[AudioToolBox] [AudioUnit] Add support for Xcode 11 beta 1 & beta 2.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Diffs are:

AudioToolBox

* iOS
  - [Beta 1](https://github.com/xamarin/xamarin-macios/wiki/AudioToolbox-iOS-xcode11-beta1)
  - [Beta 2](https://github.com/xamarin/xamarin-macios/wiki/AudioToolbox-iOS-xcode11-beta2)
* tvOS 
  - [Beta 2](https://github.com/xamarin/xamarin-macios/wiki/AudioToolbox-tvOS-xcode11-beta2)
* Mac OS X
  - [Beta 1](https://github.com/xamarin/xamarin-macios/wiki/AudioToolbox-macOS-xcode11-beta1) 
  - [Beta 2](https://github.com/xamarin/xamarin-macios/wiki/AudioToolbox-macOS-xcode11-beta2)

AudioUnit
 * iOS
   - [Beta 1](https://github.com/xamarin/xamarin-macios/wiki/AudioUnit-iOS-xcode11-beta1)
 * Mac Os X
   - [Beta 1](https://github.com/xamarin/xamarin-macios/wiki/AudioUnit-macOS-xcode11-beta1)

### CODE REVIEW

File: src/AudioToolbox/MusicTrack.cs [hunk 1]
```diff
@@ -263,7 +263,15 @@ public MusicPlayerStatus SetDestMidiEndpoint (MidiEndpoint endpoint)
 		{
 			return MusicTrackSetDestMIDIEndpoint (handle, endpoint == null ? MidiObject.InvalidRef : endpoint.MidiHandle);
 		}
-#endif // IOS
+
+		[DllImport (Constants.AudioToolboxLibrary)]
+		extern static /* OSStatus */ MusicPlayerStatus MusicTrackGetDestMIDIEndpoint (/* MusicTrack */ IntPtr inTrack, MidiEndpointRef outEndpoint);
+
+		public MusicPlayerStatus GetDestMIDIEndpoint (MidiEndpoint endpoint)
```
Comment (dalexsoto) [inline src/AudioToolbox/MusicTrack.cs]: I think you want to:

`GetDestMIDIEndpoint` => `GetDestMidiEndpoint`

Comment (mandel-macaque) [reply src/AudioToolbox/MusicTrack.cs]: missed that one, will update as soon as I get more reviews

File: src/AudioToolbox/AudioFile.cs [hunk 2]
```diff
@@ -213,6 +216,44 @@ public struct AudioFileMarker
...
+		public long IndependentlyDecodablePacket;
+	}
+
+	[Introduced (PlatformName.UIKitForMac, 13,0)]
+	[NoWatch, iOS (13,0), Mac (10,15, onlyOn64 : true), TV (13,0)]
+	[StructLayout (LayoutKind.Sequential)]
+	public struct AudioPacketDependencyInfoTranslation
+	{
+		public long Packet;
+		public uint IsIndependentlyDecodable;
```
Comment (spouliot) [inline src/AudioToolbox/AudioFile.cs]: it's a 4 bytes bool `1 means that the specified packet is independently decodable; 0 means it's not`
you can `[Marshal]` it to be nicer :)

Comment (mandel-macaque) [reply src/AudioToolbox/AudioFile.cs]: Nice catch, will be fixed.

File: src/AudioToolbox/MusicTrack.cs [hunk 3]
```diff
@@ -263,7 +263,15 @@ public MusicPlayerStatus SetDestMidiEndpoint (MidiEndpoint endpoint)
 		{
 			return MusicTrackSetDestMIDIEndpoint (handle, endpoint == null ? MidiObject.InvalidRef : endpoint.MidiHandle);
 		}
-#endif // IOS
+
+		[DllImport (Constants.AudioToolboxLibrary)]
+		extern static /* OSStatus */ MusicPlayerStatus MusicTrackGetDestMIDIEndpoint (/* MusicTrack */ IntPtr inTrack, MidiEndpointRef outEndpoint);
```
Comment (rolfbjarne) [inline src/AudioToolbox/MusicTrack.cs]: I think it should be `out MidiEndpointRef outEndpoint`.

File: src/AudioToolbox/AudioFile.cs [hunk 4]
```diff
@@ -213,6 +216,45 @@ public struct AudioFileMarker
 		}
 	}
 
+	[Introduced (PlatformName.UIKitForMac, 13,0)]
+	[NoWatch, iOS (13,0), Mac (10,15, onlyOn64 : true), TV (13,0)]
+	[StructLayout (LayoutKind.Sequential)]
+	public struct AudioPacketRangeByteCountTranslation
+	{
```
Comment (tj-devel709) [inline src/AudioToolbox/AudioFile.cs]: I believe that the opening curly braces should not be on the new line except for with methods

**Comment by mandel-macaque** [comment]
@rolfbjarne @tj-devel709 already addressed your comments.

File: src/AudioToolbox/MusicTrack.cs [hunk 5]
```diff
@@ -263,7 +263,18 @@ public MusicPlayerStatus SetDestMidiEndpoint (MidiEndpoint endpoint)
...
 		}
-#endif // IOS
+
+		[DllImport (Constants.AudioToolboxLibrary)]
+		extern static /* OSStatus */ MusicPlayerStatus MusicTrackGetDestMIDIEndpoint (/* MusicTrack */ IntPtr inTrack, out MidiEndpointRef outEndpoint);
+
+		public MusicPlayerStatus GetDestMidiEndpoint (out MidiEndpoint outEndpoint)
+		{
+			MidiEndpointRef midiHandle; 
+			var reuslt = MusicTrackGetDestMIDIEndpoint (handle, out midiHandle);
```
Comment (rolfbjarne) [inline src/AudioToolbox/MusicTrack.cs]:
```suggestion
			var result = MusicTrackGetDestMIDIEndpoint (handle, out midiHandle);
```

File: src/AudioToolbox/MusicTrack.cs [hunk 6]
```diff
@@ -263,7 +263,18 @@ public MusicPlayerStatus SetDestMidiEndpoint (MidiEndpoint endpoint)
...
+			outEndpoint = new MidiEndpoint (midiHandle);
+			return reuslt;
```
Comment (rolfbjarne) [inline src/AudioToolbox/MusicTrack.cs]:
```suggestion
			return result;
```

File: src/AudioToolbox/MusicTrack.cs [hunk 7]
```diff
@@ -263,7 +263,18 @@ public MusicPlayerStatus SetDestMidiEndpoint (MidiEndpoint endpoint)
...
-#endif // IOS
+
+		[DllImport (Constants.AudioToolboxLibrary)]
+		extern static /* OSStatus */ MusicPlayerStatus MusicTrackGetDestMIDIEndpoint (/* MusicTrack */ IntPtr inTrack, out MidiEndpointRef outEndpoint);
+
+		public MusicPlayerStatus GetDestMidiEndpoint (out MidiEndpoint outEndpoint)
+		{
+			MidiEndpointRef midiHandle; 
+			var reuslt = MusicTrackGetDestMIDIEndpoint (handle, out midiHandle);
+			outEndpoint = new MidiEndpoint (midiHandle);
```
Comment (rolfbjarne) [inline src/AudioToolbox/MusicTrack.cs]: Maybe check if the P/Invoke succeeded before creating the MidiEndpoint instance.

File: tests/monotouch-test/AudioToolbox/MusicTrackTest.cs [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+using MonoTouch.ObjCRuntime;
+using MonoTouch.CoreMidi;
+#endif
+using NUnit.Framework;
+
+namespace MonoTouchFixtures.AudioToolbox {
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class MusicTractTest {
```
Comment (rolfbjarne) [inline tests/monotouch-test/AudioToolbox/MusicTrackTest.cs]:
```suggestion
	public class MusicTrackTest {
```

File: tests/monotouch-test/AudioToolbox/MusicTrackTest.cs [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+		[SetUp]
+		public void SetUp ()
+		{
+			sequence = new MusicSequence ();
+			track = MusicTrack.FromSequence (sequence);
+		}
+
+		[TearDown]
+		public void TeatDown ()
```
Comment (rolfbjarne) [inline tests/monotouch-test/AudioToolbox/MusicTrackTest.cs]:
```suggestion
		public void TearDown ()
```

**Pull request merged**